pkg/api/endpoints.go

```diff
@@ -151,6 +151,7 @@ var configEndpoints = []API{
 		URLPath:             "/api/config/v1/dashboards/{SCOPE}/shareSettings",
 		Parent:              &dashboardAPI,
 		SingleConfiguration: true,
+		NonDeletable:        true,
```

Comment:
as a followup, could we introduce this in the regular delete please to replace the current if specifically checking for the dashboard-share-settings API.